Create datatype_tabular.md (#3333)
galaxyproject/training-material: faqs/galaxy/datatype_tabular.md

---
title: Identifying and formatting Tabular Datasets 
area: Datatypes
description: Format help for Tabular/BED/Interval Datasets
layout: faq          
---
 

A Tabular datatype is human readable and has tabs separating data columns. Please note that tabular data is different from comma seperated data (.csv) and the common datatypes are: `.bed`, `.gtf`, `.interval`, or `.txt`.
1. Click the pencil icon {% icon galaxy-pencil %} to reach the **_Edit Attributes_** form. 
   1. Change the datatype (3rd tab) and save.
   2. Label columns (1st tab) and save.
   3. Metadata will be assigned, then the dataset can be used.
2. If the required input is a BED or Interval datatype, adjusting (``.tab`` → ``.bed``, ``.tab`` → ``.interval``) maybe possible using a combination of **_Text Manipulation_** tools, to create a dataset that matches required specifications.
3. Some tools require that BED format be followed, even if the datatype Interval (with less strict column ordering) is accepted on the tool form.

>    - These tools will fail, if they are run with malformed BED datasets or non-specific column assignments.
>    - **Solution** is to reorganize the data to be in BED format and rerun. 

Reviewer: ```suggestion
   - These tools will fail, if they are run with malformed BED datasets or non-specific column assignments.
   - **Solution**: reorganize the data to be in BED format and rerun. 
```

---
title: Identifying and formatting Tabular Datasets 
area: Datatypes
description: Format help for Tabular/BED/Interval Datasets
contributors: [jennaj, beachyesh]

> layout: faq          

Reviewer: I'm not sure the order of the rows here matters, but just for consistency, could you move the contributors to the end? I think that's how it appears in the other FAQs. Thanks!

Author: Done. Thanks!

---

> title: Indentifying and Formatting Tabular Dataset 

Reviewer: ```suggestion
title: Identifying and formatting Tabular Datasets
```

Reviewer: @Beachyesh this commit was not done completely, do you know that you can do it directly from this comment here by clicking on "Commit suggestion" instead of editing the file?

area: Datatypes
description: Format help for Tabular/BED/Interval Datasets
layout: faq          
---
 

- A Tabular datatype is human readable and has tabs separating data columns.
  - Note: tabular data is different from comma seperated data (.csv)

> 1. Common tabular datatypes are `.bed`, `.gtf`, `.interval`, or `.txt`.

Reviewer: I think that this is a clarification, but not a proper step, same with step 2

Author: Noted.

> 2. The datatype metadata attribute can be directly reassigned to tabular format data.

Reviewer: I have a hard time understanding this sentence, could you rephrase it?

> 3. Click the {% icon galaxy-pencil %} ✏️ icon to reach the **_Edit Attributes_** form. 

Reviewer: This `{% icon galaxy-pencil %}` translates into ✏️, so there's no need to add ✏️

> 3. Click the {% icon galaxy-pencil %} ✏️ icon to reach the **_Edit Attributes_** form. 

Reviewer 2: We try to always also include the text description of the icon, followed by the icon itself, for accessibility reasons

```suggestion
3. Click the pencil icon {% icon galaxy-pencil %} to reach the **_Edit Attributes_** form. 
```

   - Change the datatype (3rd tab) and save

>    - label columns (1st tab) and save

Reviewer: ```suggestion
   - Label columns (1st tab) and save
```

>    -  Metadata will be assigned, then the dataset can be used.

Reviewer: ```suggestion
   - Metadata will be assigned, then the dataset can be used.
```

>    - Metadata will be assigned, then the dataset can be used.

Reviewer: these 3 could maybe be converted into steps

Author: Done.

---

> title: Datatype Tabular

Reviewer: Could we have a title here that indicates what this FAQ is trying to address?

area: Datatypes
description: Format help for Tabular/BED/Interval Datasets
layout: faq          
---
 

> - A Tabluar datatype is human readable and has tabs seperating data columns.

Reviewer: ```suggestion
- A Tabular datatype is human readable and has tabs separating data columns.
```

  - Note: tabular data is different from comma seperated data (.csv)
- Common tabular datatypes are .bed, .gtf, .interval, or .txt.
- The datatype metadata attribute can be directly reassigned to tabular format data.

> - Click the {% icon galaxy-pencil %} icon to reach the **_Edit Attributes_** form. 

Reviewer: I think that this is where the FAQ starts and we could have the numbered steps

> - Click the {% icon galaxy-pencil %} icon to reach the **_Edit Attributes_** form. 

Reviewer 2: It is best if we also write out the icon description in text, followed by the icon itself, for increased accessibility (e.g. for screen readers)

```suggestion
- Click the pencil icon {% icon galaxy-pencil %} to reach the **_Edit Attributes_** form. 
```

>   - Change the datatype (3rd tab) and save, then label columns (1st tab) and save.  Metadata will assign, then the dataset can be used.

Reviewer 2: ```suggestion
  - Change the datatype (3rd tab) and save, then label columns (1st tab) and save.  Metadata will be assigned, then the dataset can be used.
```

- If the required input is a BED or Interval datatype, adjusting (.tab → .bed, .tab → .interval) maybe possible using a combination of **_Text Manipulation_** tools, to create a dataset that matches the BED or Interval datatype specifications.

#### Tips
- Some tools require that BED format is followed, even if the datatype Interval (with less strict column ordering) is accepted on the tool form.

> - These tools will fail if run with malformed BED datasets or non-specific column assignments.

Reviewer: do you think that these tips could be included within the previous bullet point?

> **Solution**: Reorganize the data to be in BED format and rerun. 

Reviewer: is this also a step in the FAQ?

---
title: Datatype Tabular

> area: Getting inputs right    # FAQs will be grouped by these areas on the FAQ page

Reviewer 2: please try to align with existing areas, this sounds more like a title

description: Format help for Tabular/BED/Interval Datasets
layout: faq          

> 

Reviewer 2: please remove this empty line, it may break the rendering

---
 

> ### Help for Tabular Datasets

Reviewer 2: you don't need a title here, the title defined in the metadata above will be placed at the top of the FAQ

---
title: Datatype Tabular
area: Getting inputs right    # FAQs will be grouped by these areas on the FAQ page
description: Format help for Tabular/BED/Interval Datasets
layout: faq          
contributors: [beachyesh] 
---

> [Back to Support Hub](https://galaxyproject.org/support/)

Reviewer 3: please do not include this.

Related Topics 
- [Fastq Manipulation and Quality Control](https://galaxyproject.org/tutorials/ngs/#fastq-manipulation-and-quality-control)  
- [How to format fastq data for tools that require .fastqsanger format?](https://galaxyproject.org/support/fastqsanger/)
- [Understanding compressed fastq data (fastq.gz)](https://galaxyproject.org/support/compressed-fastq/)
- [Common datatypes explained](https://galaxyproject.org/learn/datatypes/)

> - [Input datatype misassignment and errors](https://galaxyproject.org/support/tool-error/)

Reviewer 3: this is interesting but I'm not sure we should include this especially when e.g. `Understanding compressed fastq data (fastq.gz)` will probably become a faq sometday.

I would suggest removing this
